include/swift/AST/Evaluator.h

```diff
@@ -208,6 +208,18 @@ class Evaluator {
   /// is treated as a stack and is used to detect cycles.
   llvm::SetVector<ActiveRequest> activeRequests;
 
+  /// How many `ResolveMacroRequest` requests are active.
+  ///
+  /// This allows us to quickly determine whether there is any
+  /// `ResolveMacroRequest` active in the active request stack.
+  /// It saves us from a linear scan through `activeRequests` when
+  /// we need to determine this information.
```

Comment:
`activeRequests` is a `SetVector`, so it wouldn't be a linear scan, right? (at least only in the absolute worst case)

Reply:
The hash function for requests includes both the request kind and the payload. In this case, I need to know if there is *any* `ResolveMacroRequest` on the stack (regardless of payload), not whether there is a *specific* instance of `ResolveMacroRequest`. Now, we could probably do something tricky where the `ResolveMacroRequest` hash function is constant (so only the type-id is relevant), which would allow us to use a form of lookup into the `SetVector` that only looks based on the hash value... but that seems more dodgy than what I'm doing.

Reply:
Ah I see, makes sense